src/dirent.rs

```diff
@@ -0,0 +1,40 @@
+use {Result, Error, Errno};
+use libc::{self, DIR};
+use std::os::unix::io::RawFd;
+use errno;
+
+pub struct Dir {
+    pub handle: *mut DIR,
```

Comment:
I don't think this field should be public. Instead I'd propose implementing `AsRawFd` and `IntoRawFd` instead.